src/algebra/star/subalgebra.lean

```diff
@@ -225,3 +250,341 @@ set.centralizer_subset h
+  { cases hx, exacts [Halg_right _ _ hx, star_star x ▸ Hstar_left _ _ (Halg_right r _ hx)] },
+end
+
+/-- The difference with `star_subalgebra.adjoin_induction` is that this acts on the subtype. -/
+lemma adjoin_induction' {s : set A} {p : adjoin R s → Prop} (a : adjoin R s)
+  (Hs : ∀ x (h : x ∈ s), p ⟨x, subset_adjoin R s h⟩)
+  (Halg : ∀ r, p (algebra_map R _ r)) (Hadd : ∀ x y, p x → p y → p (x + y))
+  (Hmul : ∀ x y, p x → p y → p (x * y)) (Hstar : ∀ x, p x → p (star x)) : p a :=
+subtype.rec_on a $ λ b hb,
+begin
+  refine exists.elim _ (λ (hb : b ∈ adjoin R s) (hc : p ⟨b, hb⟩), hc),
+  apply adjoin_induction R hb,
+  exacts [λ x hx, ⟨subset_adjoin R s hx, Hs x hx⟩,
+    λ r, ⟨star_subalgebra.algebra_map_mem _ r, Halg r⟩,
+    (λ x y hx hy, exists.elim hx $ λ hx' hx,
+      exists.elim hy $ λ hy' hy, ⟨add_mem hx' hy', Hadd _ _ hx hy⟩),
+    (λ x y hx hy, exists.elim hx $ λ hx' hx,
+      exists.elim hy $ λ hy' hy, ⟨mul_mem hx' hy', Hmul _ _ hx hy⟩),
+    λ x hx, exists.elim hx (λ hx' hx, ⟨star_mem hx', Hstar _ hx⟩)]
+end
+
+/-- If all elements of `s : set A` commute pairwise and also commute pairwise with elements of
+`star s`, then `star_subalgebra.adjoin R s` is commutative. See note [reducible non-instances]. -/
+@[reducible]
+def adjoin_comm_semiring_of_comm {s : set A}
+  (hcomm : ∀ (a : A), a ∈ s → ∀ (b : A), b ∈ s → a * b = b * a)
+  (hcomm_star : ∀ (a : A), a ∈ s → ∀ (b : A), b ∈ s → a * star b = star b * a) :
+comm_semiring (adjoin R s) :=
+{ mul_comm :=
+  begin
+    rintro ⟨x, hx⟩ ⟨y, hy⟩,
+    ext,
+    simp only [set_like.coe_mk, mul_mem_class.mk_mul_mk],
+    rw [←mem_to_subalgebra, adjoin_to_subalgebra] at hx hy,
+    letI : comm_semiring (algebra.adjoin R (s ∪ star s)) := algebra.adjoin_comm_semiring_of_comm R
+    begin
+      intros a ha b hb,
+      cases ha; cases hb,
+      exacts [hcomm _ ha _ hb, star_star b ▸ hcomm_star _ ha _ hb,
+        star_star a ▸ (hcomm_star _ hb _ ha).symm,
+        by simpa only [star_mul, star_star] using congr_arg star (hcomm _ hb _ ha)],
+    end,
+    exact congr_arg coe (mul_comm (⟨x, hx⟩ : algebra.adjoin R (s ∪ star s)) ⟨y, hy⟩),
+  end,
+  ..(adjoin R s).to_subalgebra.to_semiring }
+
+/-- If all elements of `s : set A` commute pairwise and also commute pairwise with elements of
+`star s`, then `star_subalgebra.adjoin R s` is commutative. See note [reducible non-instances]. -/
+@[reducible]
+def adjoin_comm_ring_of_comm (R : Type u) {A : Type v} [comm_ring R] [star_ring R]
+  [ring A] [algebra R A] [star_ring A] [star_module R A] {s : set A}
+  (hcomm : ∀ (a : A), a ∈ s → ∀ (b : A), b ∈ s → a * b = b * a)
+  (hcomm_star : ∀ (a : A), a ∈ s → ∀ (b : A), b ∈ s → a * star b = star b * a) :
+comm_ring (adjoin R s) :=
+{ ..star_subalgebra.adjoin_comm_semiring_of_comm R hcomm hcomm_star,
+  ..(adjoin R s).to_subalgebra.to_ring }
+
+/-- The star subalgebra `star_subalgebra.adjoin R {x}` generated by a single `x : A` is commutative
+in the presence of `is_star_normal x`. -/
```

Comment:
```suggestion
if `x` is normal. -/
```
This means no clickable name, but I think it is more readable, and even if the reader doesn't know what "normal" means in this context it will be pretty obvious to look into the `is_star_normal` hypothesis